sdk/python/feast/entity.py

```diff
@@ -44,3 +56,246 @@ def from_proto(cls, entity_proto: EntityProto):
         """
         entity = cls(name=entity_proto.name, dtype=ValueType(entity_proto.value_type))
         return entity
+
+
+class EntityV2:
+    """
+    Represents a collection of entities and associated metadata.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        value_type: ValueType,
+        labels: Optional[MutableMapping[str, str]] = None,
+    ):
+        self._name = name
+        self._description = description
+        self._value_type = value_type
+        if labels is None:
+            self._labels = OrderedDict()  # type: MutableMapping[str, str]
+        else:
+            self._labels = labels
+
+        self._created_timestamp: Optional[Timestamp] = None
+
+    def __eq__(self, other):
+        if not isinstance(other, EntityV2):
+            return NotImplemented
```

Comment:
Should this not be a descriptive exceptioN?

Reply:
Added.